samples/snippets/csharp/VS_Snippets_CLR/Formatting.HowTo.RoundTrip/cs/RoundTrip.cs

```diff
@@ -4,120 +4,128 @@
+
+        // Save DateTime value.
+        DateTimeOffset dateToSave = new DateTimeOffset(2008, 6, 12, 18, 45, 15,
+                                                       new TimeSpan(7, 0, 0));
+        string? dateString = dateToSave.ToString("o");
+        Console.WriteLine("Converted {0} to {1}.", dateToSave.ToString(),
+                          dateString);
+        outFile.WriteLine(dateString);
+        Console.WriteLine("Wrote {0} to {1}.", dateString, fileName);
+        outFile.Close();
+
+        // Restore DateTime value.
+        DateTimeOffset restoredDateOff;
+
+        using StreamReader inFile = new StreamReader(fileName);
+        dateString = inFile.ReadLine();
+
+        if (dateString != null)
+        {
+            restoredDateOff = DateTimeOffset.Parse(dateString, null,
+                                                   DateTimeStyles.RoundtripKind);
+            Console.WriteLine("Read {0} from {1}.", restoredDateOff.ToString(), fileName);
+        }
+
+        // The example displays the following output:
+        //    Converted 6/12/2008 6:45:15 PM +07:00 to 2008-06-12T18:45:15.0000000+07:00.
+        //    Wrote 2008-06-12T18:45:15.0000000+07:00 to .\DateOff.txt.
+        //    Read 6/12/2008 6:45:15 PM +07:00 from .\DateOff.txt.
+        // </Snippet2>
+    }
 }
 
 // <Snippet3>
-[Serializable] public class DateInTimeZone
+[Serializable]
+public class DateInTimeZone
 {
-   private TimeZoneInfo tz;
-   private DateTimeOffset thisDate;
-
-   public DateInTimeZone() {}
-
-   public DateInTimeZone(DateTimeOffset date, TimeZoneInfo timeZone)
-   {
-      if (timeZone == null)
-         throw new ArgumentNullException("timeZone", "The time zone cannot be null.");
-
-      this.thisDate = date;
-      this.tz = timeZone;
-   }
-
-   public DateTimeOffset DateAndTime
-   {
-      get {
-         return this.thisDate;
-      }
-      set {
-         if (value.Offset != this.tz.GetUtcOffset(value))
-            this.thisDate = TimeZoneInfo.ConvertTime(value, tz);
-         else
-            this.thisDate = value;
-      }
-   }
-
-   public TimeZoneInfo TimeZone
-   {
-      get {
-         return this.tz;
-      }
-   }
+    private TimeZoneInfo? tz;
+    private DateTimeOffset thisDate;
+
+    public DateInTimeZone() { }
+
+    public DateInTimeZone(DateTimeOffset date, TimeZoneInfo timeZone)
+    {
+        if (timeZone == null)
+            throw new ArgumentNullException("timeZone", "The time zone cannot be null.");
+
+        this.thisDate = date;
+        this.tz = timeZone;
+    }
+
+    public DateTimeOffset DateAndTime
+    {
+        get
+        {
+            return thisDate;
+        }
+        set
+        {
+            if ((tz is not null) && (value.Offset != tz.GetUtcOffset(value)))
+                thisDate = TimeZoneInfo.ConvertTime(value, tz);
+            else
+                thisDate = value;
+        }
+    }
+
+    public TimeZoneInfo? TimeZone
+    {
+        get => tz;
+    }
```

Comment:
I think this can be simplified:

```suggestion
    public TimeZoneInfo? TimeZone => tz;
```